BookingQueueSubscriber/BookingQueueSubscriber.Services/UserApi/UserService.cs

```diff
@@ -37,20 +38,48 @@ public UserService(IUserApiClient userApiClient, ILogger<UserService> logger, IF
             _featureToggles = featureToggles;
         }
 
-        public async Task<User> CreateNewUserForParticipantAsync(string firstname, string lastname, string contactEmail, bool isTestUser)
+        public async Task<User> CreateNewUserForParticipantAsync(string firstname, string lastname, string contactEmail,
+            bool isTestUser)
         {
             var userProfile = await GetUserByContactEmail(contactEmail);
             if (userProfile == null)
             {
-                _logger.LogInformation("User with contact email {contactEmail} does not exist. Creating an account.", contactEmail);
-                // create the user in AD.
-                var newUser = await CreateNewUserInAD(firstname, lastname, contactEmail, isTestUser);
-                return new User
+                _logger.LogInformation("User with contact email {contactEmail} does not exist. Creating an account.",
+                    contactEmail);
+
+                try
                 {
-                    UserId = newUser.UserId,
-                    UserName = newUser.Username,
-                    Password = newUser.OneTimePassword
-                };
+                    // create the user in AD.
+                    var newUser = await CreateNewUserInAD(firstname, lastname, contactEmail, isTestUser);
+                    
+                    
+
+                    return new User
+                    {
+                        UserId = newUser.UserId,
+                        UserName = newUser.Username,
+                        Password = newUser.OneTimePassword
+                    };
+                }
+                catch (UserApiException e)
+                {
+                    if (e.StatusCode == (int) HttpStatusCode.Conflict)
+                    {
+                        Thread.Sleep(1000);
+                        userProfile = await GetUserByContactEmail(contactEmail);
+                        if (userProfile == null)
+                        {
+                            _logger.LogInformation(
+                                "User with contact email {contactEmail} does not exist. Creating an account. Second try.",
+                                contactEmail);
+                            return null;
+                        }
+                    }
+                    else
+                    {
+                        return null;
```

Comment:
I think we said we were going to use `throw` here in order to preserve the original exception. Returning null here will cause a null reference exception for the caller when it tries to set the username, which would mask the original error.

`participant.Username = user.UserName;`

Reply:
I changed the parent and it checks for null. 
On UserCreationAndNotification.cs in CreateUserAndSendNotificationAsync it is already checking for user null from the beginning. I thought that it is better to be consistent in the method. What I am saying is that it is already returning a null user in the original code.

Reply:
oops sorry didn't see that you'd changed the parent, my bad